flow/src/lib.rs

```diff
@@ -836,6 +733,171 @@ impl<T: Config> Pallet<T> {
 		Nonce::<T>::put(nonce.wrapping_add(1));
 		nonce
 	}
+
+	fn schedule_campaign_settlements(block_number: T::BlockNumber) {
+		// Iterate over campaigns ending in this block
+		for campaign_id in &CampaignsByBlock::<T>::get(block_number) {
+			let campaign = Campaigns::<T>::get(campaign_id);
+			let campaign_balance = CampaignBalance::<T>::get(campaign_id);
+			
+			// Campaign cap reached: Finalizing
+			if campaign_balance >= campaign.cap {
+				Self::set_state(campaign.id, FlowState::Finalizing);
+
+				Self::deposit_event(Event::CampaignFinalising {
+					campaign_id: *campaign_id,
+					campaign_balance,
+					block_number,
+					success: false,
```

Comment:
What `success` flag means in both `CampaignFinalising` and `CampaignReverting` events and in which scenarios is it possible for this value to be true?

Reply:
It should be `true` only when the finalization process is finished with the status `Finalized`. Not sure if it's needed at all for these Events. I've added it only to be generic with `CampaignFailed` and `CampaignFinalized`.
I think we can drop `success` flag for `CampaignFinalising` and `CampaignReverting`.

Reply:
Fixed.